src/main/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLogger.java

```diff
@@ -259,19 +260,25 @@ public MetricsLogger setTimestamp(Instant timestamp) throws InvalidTimestampExce
         return this;
     }
 
-    @SneakyThrows
     private void configureContextForEnvironment(MetricsContext context, Environment environment) {
         if (context.hasDefaultDimensions()) {
             return;
         }
         DimensionSet defaultDimension = new DimensionSet();
-        defaultDimension.addDimension("LogGroup", environment.getLogGroupName());
-        defaultDimension.addDimension("ServiceName", environment.getName());
-        defaultDimension.addDimension("ServiceType", environment.getType());
+        setDefaultDimension(defaultDimension, "LogGroup", environment.getLogGroupName());
+        setDefaultDimension(defaultDimension, "ServiceName",  environment.getName());
+        setDefaultDimension(defaultDimension, "ServiceType", environment.getType());
         context.setDefaultDimensions(defaultDimension);
         environment.configureContext(context);
     }
 
+    @SneakyThrows
+    private void setDefaultDimension(DimensionSet defaultDimension, String dimKey, String dimVal) {
+        if (!StringUtils.isNullOrEmpty(dimVal)) {
```

Comment:
Non-empty strings with only whitespace will still throw an error.
What about having an extra check for `isBlank()` ?

Reply:
I will just simply try on existing dimension validator as even we fix blank, another question can come out what if this dimensionValue has non valid ascii : )